### PR TITLE
SED-2964 compare mode for custom dashboards

### PR DESCRIPTION
**Pull request by lucianbaciu**

### CODE REVIEW

**Review by neogucky — CHANGES_REQUESTED** [review]

File: projects/step-core/src/lib/client/generated/models/AxesSettings.ts [hunk 1]
```diff
@@ -10,5 +10,5 @@ export type AxesSettings = {
   displayType: 'LINE' | 'BAR_CHART';
   unit: string;
   renderingSettings?: MetricRenderingSettings;
-  colorizationType: 'STROKE' | 'FILL';
+  colorizationType?: 'STROKE' | 'FILL';
```
[the remark this answers is not in the document]
Comment (lucianbaciu) [reply projects/step-core/src/lib/client/generated/models/AxesSettings.ts:13]: @neogucky you are right. it was defined with a default value but forgot the @NotNull annotation. will add it back

File: projects/step-frontend/src/lib/modules/timeseries/components/table-dashlet/table-dashlet.component.ts [hunk 2]
```diff
@@ -312,14 +485,26 @@ export class TableDashletComponent extends ChartDashlet implements OnInit, OnCha
...
       .addSortNumberPredicate('TPH_diff', (item) => item.tphDiff)
       .build();
   }
+
+  percentageBetween(x: number | undefined, y: number | undefined) {
+    if (x && y) {
+      return ((y - x) / x) * 100;
+    } else {
+      return undefined;
+    }
```
[the remark this answers is not in the document]
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/components/table-dashlet/table-dashlet.component.ts:502]: good point. y being 0 is a good scenario

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts [hunk 3]
```diff
@@ -206,6 +199,7 @@ export class ChartDashletComponent extends ChartDashlet implements OnInit {
   }
 
   handleLockStateChange(locked: boolean) {
+    console.log('LOCK STATE');
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts:202]:
```suggestion
```

**Review by neogucky — APPROVED** [review]
Please remove the remaining logs, afterwards a merge would be fine from my side

**Review by neogucky — APPROVED** [review]

**Pull request merged**